pkg/prometheus/resource_selector.go

```diff
@@ -762,8 +762,8 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			rejectFn(sc, err)
 			continue
 		}
-		if sc.Spec.ProxyConfig != nil {
-			if err = validateProxyConfig(ctx, sc.Spec.ProxyConfig, rs.store, sc.GetNamespace()); err != nil {
+		if monitoringv1.IsProxyConfigNotEqualToZero(sc.Spec.ProxyConfig) {
+			if err = validateProxyConfig(ctx, &sc.Spec.ProxyConfig, rs.store, sc.GetNamespace()); err != nil {
```

Comment:
(suggestion) same remark: validateProxyConfig() can receive the value instead of the pointer and check the zero value directly.